### PR TITLE
Fix Qt 6.4 deprecated declarations

### PR DESCRIPTION
**Pull request by Holzhaus**
See the commit messages for details.

### CODE REVIEW

**Review by uklotzde — APPROVED** [review]

File: src/util/imageutils.cpp [hunk 1]
```diff
@@ -15,9 +16,10 @@ ImageDigest digestImage(const QImage& image) {
         return ImageDigest();
     }
     QCryptographicHash cryptoHash(kImageHashAlgorithm);
-    cryptoHash.addData(
+    const QByteArray data = QByteArray::fromRawData(
```
Comment (Swiftb0y) [inline src/util/imageutils.cpp:19]: why not QByteArrayView?

Comment (Holzhaus) [reply src/util/imageutils.cpp:19]: `QByteArrayView` only exists in Qt 6.0+. Also, `QByteArray::fromRawData` does not create a copy either, so it should be fine.

Comment (Swiftb0y) [reply src/util/imageutils.cpp:19]: A right, I forgot this is not Qt6-only code.

File: src/library/browse/browsefeature.cpp [hunk 2]
```diff
@@ -462,9 +462,7 @@ void BrowseFeature::loadQuickLinks() {
 }
 
 QString BrowseFeature::extractNameFromPath(const QString& spath) {
-    QString path = spath.left(spath.count()-1);
-    int index = path.lastIndexOf("/");
-    QString name = (spath.count() > 1) ? path.mid(index+1) : spath;
+    return QFileInfo(spath).fileName();
     return name;
```
Comment (Swiftb0y) [inline src/library/browse/browsefeature.cpp:466]:
```suggestion
    return QFileInfo(spath).fileName();
```

**Review by Swiftb0y — COMMENTED** [review]
sorry, my previous suggestion didn't remove the outdated line...

**Review by Swiftb0y — APPROVED** [review]

**Comment by Swiftb0y** [comment]
Thank you for keeping up with Qt6.

**Pull request merged**